core/trino-main/src/main/java/io/trino/sql/planner/PlanOptimizers.java

```diff
@@ -435,8 +435,9 @@ public PlanOptimizers(
                         estimatedExchangesCostCalculator,
                         ImmutableSet.<Rule<?>>builder()
                                 .addAll(columnPruningRules)
-                                .addAll(projectionPushdownRules)
                                 .addAll(ImmutableSet.of(
+                                        new PushProjectionThroughUnion(),
```

Comment:
my understanding is that we want to converge towards being able to put all rules into this particular iterative optimizer. so I'm not sure if removing stuff from here is the right direction. 

The core issue the PR is trying to solve seems to be that connector pushdown support may be different in source and destination, so may be it's better not to change anything other than `PushProjectionIntoTableScan` invocation. 

@martint wdyt?

Reply:
The problem is that our optimizer is greedy. So depending on rule execution (e.g pushdown into connector rules), we might not have a chance to apply redirection rule. Therefore, it would be best if we could "try" redirection rule before any pushdowns.

Generally, it's odd that we do connector pushdown that early while any other connector pushdowns (predicate, aggregations, joins) is performed later.

Reply:
Made changes now to just move PushProjectionIntoTableScan